docs/v1/P4Runtime-Spec.mdk

```diff
@@ -1191,6 +1193,12 @@ encodes the object type.
 +--------------------------------+------------------------------------------------------------+
 ~
 
+The `@id` annotation can also be used to choose the ID for match fields,
+action parameters, and packet metadata. In this case, there is no 8-bit prefix
+and the programmer is free to choose any 32-bit number. The compiler must fail
+if the IDs chosen by the programmer are not unique (within a table, action, or
+header, respectively).
```

Comment:
I do not fully know the reasons for the 8-bit prefix identifying the type of object, but is allowing the most significant 8 bits of the id of a match field, action parameter, and/or packet metadata to be the same as that of one of the 'defined prefixes' in the spec actually what you want?  i.e. is anyone planning on taking an arbitrary id, selecting out its most significant 8 bits, and using that to determine what type of object it is?

Reply:
I'm not sure what the reason is (I feel having that tag there in the first place seems unnecessary), so I don't know. I personally think it is fine as it is, but I'm open to hearing others give their opinion.

Reply:
These ids are a bit specific in that their scope is limited to their parent P4 entity (e.g. table for match field parameters). That being said, it could make sense to restrict the prefix to be 0 to avoid confusion.

Reply:
I don't feel strongly either way, would people like me to restrict these IDs to 24 bit?

Reply:
In the WG meeting on 3/25 we decided not to restrict these IDs to keep the specification simple.